### PR TITLE
provider/azurerm: Add example of a VNET w/ Two Subnets

### PR DESCRIPTION
**Pull request by anniehedgpeth**
## CI: [Build Status](https://travis-ci.org/harijayms/terraform)

This example provisions and deploys the following resources into an Azure subscription:

- Resource Group
- VNet
- Subnet 1
- Subnet 2

```
terraform get, validate, plan to output, apply output, show
validate with azure CLI
terraform destroy
```

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -1,42 +1,36 @@
...
-sudo: false
-language: go
-go:
-- 1.8
-
-# add TF_CONSUL_TEST=1 to run consul tests
-# they were causing timouts in travis
+####################################################################################
+## NOT FOR UPSTREAM PROPOSAL; INTENDED FOR CI OF AZURE EXAMPLES IN THIS REPO ONLY ##
+####################################################################################
```
Comment (tombuildsstuff) [inline .travis.yml]: can we revert this change? :)

File: examples/azure-vm-simple-linux-managed-disk/README.md [hunk 2]
```diff
@@ -0,0 +1,24 @@
+# Very simple deployment of a Linux VM
+
+<a href="http://armviz.io/#/?load=https%3A%2F%2Fraw.githubusercontent.com%2FAzure%2Fazure-quickstart-templates%2Fmaster%2F101-vm-simple-linux%2Fazuredeploy.json" target="_blank">
+    <img src="http://armviz.io/visualizebutton.png"/>
+</a>
```
Comment (tombuildsstuff) [inline examples/azure-vm-simple-linux-managed-disk/README.md]: can we remove this, seeing as it's linking to a separate ARM Template (which could get out of sync)?

File: examples/azure-vm-simple-linux-managed-disk/README.md [hunk 3]
```diff
@@ -0,0 +1,24 @@
...
+
+This template allows you to deploy a simple Linux VM using a few different options for the Ubuntu version, using the latest patched version. This will deploy a A1 size VM in the resource group location and return the FQDN of the VM.
```
Comment (tombuildsstuff) [inline examples/azure-vm-simple-linux-managed-disk/README.md]: [Based on this](https://github.com/hashicorp/terraform/pull/14115/files#diff-3ccea4997df6993ff3a98e3548cf9174R45) this will spin up an A0 - can we update this to match?

File: examples/azure-vm-simple-linux-managed-disk/provider.tf [hunk 4]
```diff
@@ -0,0 +1,7 @@
+# provider "azurerm" {
+#   subscription_id = "REPLACE-WITH-YOUR-SUBSCRIPTION-ID"
+#   client_id       = "REPLACE-WITH-YOUR-CLIENT-ID"
+#   client_secret   = "REPLACE-WITH-YOUR-CLIENT-SECRET"
+#   tenant_id       = "REPLACE-WITH-YOUR-TENANT-ID"
+# }
```
Comment (tombuildsstuff) [inline examples/azure-vm-simple-linux-managed-disk/provider.tf]: can we move this into the `main.tf` file?

File: examples/azure-vm-simple-linux-managed-disk/terraform.tfvars [hunk 5]
```diff
@@ -0,0 +1,8 @@
+# Replace with relevant values
+
+# resource_group = "myresourcegroup"
+# rg_prefix      = "rg"
+# hostname       = "myvm"
+# dns_name       = "mydnsname"
+# location       = "southcentralus"
+# admin_password = "T3rr@f0rmP@ssword"
```
Comment (tombuildsstuff) [inline examples/azure-vm-simple-linux-managed-disk/terraform.tfvars]: can we move the provider config into the `main.tf` file?

File: examples/azure-vnet-two-subnets/provider.tf.example [hunk 6]
```diff
@@ -0,0 +1,6 @@
+provider "azurerm" {
+  subscription_id = "REPLACE-WITH-YOUR-SUBSCRIPTION-ID"
+  client_id       = "REPLACE-WITH-YOUR-CLIENT-ID"
+  client_secret   = "REPLACE-WITH-YOUR-CLIENT-SECRET"
+  tenant_id       = "REPLACE-WITH-YOUR-TENANT-ID"
+}
```
Comment (tombuildsstuff) [inline examples/azure-vnet-two-subnets/provider.tf.example]: can we move this into the other `main.tf` too?

File: .travis.yml [hunk 7]
```diff
@@ -25,7 +25,7 @@ install:
 - bash scripts/gogetcookie.sh
 - go get github.com/kardianos/govendor
 script:
-- make vendor-status test vet
+- make vet vendor-status test
```
Comment (tombuildsstuff) [inline .travis.yml]: (I've fixed the merge conflict here and reverted this back to how it was in `96617b609bf7abd0979b9c099b91cc79feac4461`)

**Comment by anniehedgpeth** [comment]
Thanks @tombuildsstuff !
